packages/creator-hub/renderer/src/modules/store/deployment/slice.ts

```diff
@@ -214,11 +236,23 @@ const deploymentSlice = createSlice({
   initialState,
   reducers: {
     updateDeploymentStatus: (state, action: PayloadAction<UpdateDeploymentStatusPayload>) => {
-      const { path, componentsStatus } = action.payload;
+      const { path, deploymentId, componentsStatus } = action.payload;
       const deployment = state.deployments[path];
-      if (deployment) {
+
+      // Only update if the deployment ID matches (prevents updating wrong deployment)
+      if (deployment && deployment.id === deploymentId) {
         deployment.componentsStatus = componentsStatus;
         deployment.lastUpdated = Date.now();
+      } else {
+        // If deployment was moved to history, update it there
+        const historyDeployments = state.history[path];
+        if (historyDeployments) {
+          const historyDeployment = historyDeployments.find(d => d.id === deploymentId);
+          if (historyDeployment) {
+            historyDeployment.componentsStatus = componentsStatus;
+            historyDeployment.lastUpdated = Date.now();
+          }
+        }
```

Comment:
is ok to be updating history deployments? feels like we should push to history only the already finished ones (readonly)

Reply:
as the deployments could take at least 15 min, this logic is to keep fetching the AB and LOD status